pkg/cmd/roachtest/tests/mixed_version_backup.go

```diff
@@ -1359,19 +1360,30 @@ func (mvb *mixedVersionBackup) backupName(
 // error if the job doesn't succeed within the attempted retries.
 func (mvb *mixedVersionBackup) waitForJobSuccess(
 	ctx context.Context, l *logger.Logger, rng *rand.Rand, h *mixedversion.Helper, jobID int,
-) error {
+) (resErr error) {
 	var lastErr error
-	node, db := h.RandomDB(rng, mvb.roachNodes)
+	node := h.RandomNode(rng, mvb.roachNodes)
 	l.Printf("querying job status through node %d", node)
 
+	db, err := mvb.cluster.ConnE(ctx, l, node, option.DBName("system"))
+	if err != nil {
+		l.Printf("error connecting to node %d: %v", node, err)
+		return err
+	}
+	defer func() {
+		err := db.Close()
+		resErr = errors.CombineErrors(resErr, err)
+	}()
```

Comment:
FWIW, the `system` connections could be created once `OnStartup` and just referenced directly in this function, avoiding creating a new connection pool every time this function is called and simplifying this code by removing this error handling logic. Up to you.